File: miraculous-frontend/miraculous-ui/src/dto/track-like-dto/track-like-dto.ts
export class TrackLikeDto {

  siteVisitorId: number;
  trackId: number;
  numberOfLikes: number;
  trackLikedByCurrentVisitor: boolean;


  constructor(siteVisitorId?: number, trackId?: number, numberOfLikes?: number, isTrackLikedByCurrentVisitor?: boolean) {
    this.siteVisitorId = siteVisitorId;
    this.trackId = trackId;
    this.numberOfLikes = numberOfLikes;
    this.trackLikedByCurrentVisitor = isTrackLikedByCurrentVisitor;
  }
}

File: miraculous-backend/src/main/java/com/example/miraculousbackend/dto/TrackLikeDto.java
package com.example.miraculousbackend.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TrackLikeDto {

    private Long siteVisitorId;
    private Long trackId;
    private long numberOfLikes;
    private boolean isTrackLikedByCurrentVisitor;
}
